### DISCUSSION TITLE
zh/integrations/tfjs/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17759] [Docs]
# zh/integrations/tfjs/

将您的Ultralytics YOLO11  模型转换为TensorFlow.js，以进行高速本地对象检测。了解如何为浏览器和 Node.js 应用程序优化 ML 模型。

https://docs.ultralytics.com/zh/integrations/tfjs/

**Comment by lilinebo** [comment]
Hello author, I am a newbie. I trained a yolov11n OBB detection model and then converted it to tfjs format. I found that the pt model inference result is an object, so how to parse the result in tfjs?

**Reply by glenn-jocher** [reply]
@mozeqiu123 to parse the inference results in TensorFlow.js, check the format of the output from the exported YOLO model. Typically, it should provide arrays of tensor data. Map the output tensors to expected object properties like bounding box coordinates, class IDs, and confidence scores in your application logic to process and display the results. Let me know if you need more details!

**Reply by lilinebo** [reply]
Thank you for your reply, the specific information is here：https://github.com/ultralytics/ultralytics/issues/17768

**Reply by glenn-jocher** [reply]
Thank you for sharing the reference issue. To parse YOLO11 OBB detection results in TensorFlow.js, you will need to interpret the model's output tensors correctly based on your trained model's format. Typically, the outputs include bounding box coordinates, class predictions, confidence scores, and oriented box angles. You can check the model's documentation or inspect the export code to map these outputs properly. Let us know if further clarification is needed!

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! It looks like you are exploring the exciting possibility of converting Ultralytics YOLO models to TensorFlow.js. For detailed guidance, you might want to check out our [integration documentation](https://docs.ultralytics.com/zh/integrations/tfjs/).

If this is a 🐛 Bug Report regarding the conversion process or similar issues, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and resolve your issue.

In case you have any specific questions or face challenges related to custom training ❓, please share additional details, such as code snippets or logs. Don't forget to follow our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your model training.

Join our vibrant Ultralytics community for discussions and help:
- For quick responses, hop on to our [Discord](https://ultralytics.com/discord) 🎧.
- For detailed discussions, visit [Discourse](https://community.ultralytics.com).
- If you prefer community threads, check out our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

It's always a good idea to ensure you're using the latest version of `ultralytics`. Update your package, along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment using [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO models in any of these verified environments with all dependencies, including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/), preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Our CI tests, shown by the badge above, ensure all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) function correctly across platforms like macOS, Windows, and Ubuntu. These tests run every 24 hours and on each commit to guarantee reliability.

You will receive support from an Ultralytics engineer soon. 😊
